cli/command/node/health.go

```diff
@@ -78,3 +82,52 @@ func runHealth(storageosCli *command.StorageOSCli, opt *healthOptions) error {
 	}
 	return formatter.NodeHealthWrite(nodeHealthCtx, node)
 }
+
+// UpdateNodeHealth updates the health status of a given node by querying the
+// node endpoints.
+func UpdateNodeHealth(node *cliTypes.Node, address string, timeout int) error {
+	const healthEndpointFormat = "http://%s:%s/v1/health"
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(timeout))
+	defer cancel()
+
+	client := &http.Client{}
+
+	// CP Health Status
+	var cpStatus *types.CPHealthStatus
+	cpURL := fmt.Sprintf(healthEndpointFormat, address, DefaultPort)
+	cpReq, err := http.NewRequest("GET", cpURL, nil)
+	if err != nil {
+		return err
+	}
+
+	cpResp, err := client.Do(cpReq.WithContext(ctx))
+	if err != nil {
+		return err
+	}
+
+	if err := json.NewDecoder(cpResp.Body).Decode(&cpStatus); err != nil {
```

Comment:
cpStatus is a pointer

Reply:
Changed it to a non-pointer above. Thanks!